blockstore/splitstore/splitstore_compact.go

```diff
@@ -926,11 +934,19 @@ func (s *SplitStore) walkChain(ts *types.TipSet, inclState, inclMsgs abi.ChainEp
 		err = s.view(c, func(data []byte) error {
 			return hdr.UnmarshalCBOR(bytes.NewBuffer(data))
 		})
-
 		if err != nil {
 			return xerrors.Errorf("error unmarshaling block header (cid: %s): %w", c, err)
 		}
 
+		// tipset CID references are retained
+		if pRef, err := tsRef(hdr.Parents); err != nil {
+			return xerrors.Errorf("error computing cid reference to parent tipset")
+		} else {
+			if err := s.walkObjectIncomplete(pRef, visitor, fHot, stopWalk); err != nil {
```

Comment:
Seems likely tipsets produced before nv18 won't retroactively get references added to the store so this is an incomplete walk.